lib/driver.js

```diff
@@ -184,8 +194,29 @@ class BaseDriver extends MobileJsonWireProtocol {
 
     if (!_.contains(validStrats, strategy)) {
       throw new errors.InvalidSelectorError(`Locator Strategy "${strategy}" is not supported for this session`);
+    }
+  }
+
+  /*
+   * Restart the session with the original caps,
+   * preserving the timeout config.
+   */
+  async reset () {
+    log.debug("Resetting app mid-session");
+    log.debug("Running generic full reset");
+    let currentConfig = {};
+    for (let k of ['implicitWaitMs', 'newCommandTimeoutMs', 'sessionId']) {
+      currentConfig[k] = this[k];
+    }
+    await this.deleteSession();
+    log.debug("Restarting app");
+    await this.createSession(this.caps);
+
+    for (let [k,v] of _.pairs(currentConfig)) {
+      this[k] = v;
```

Comment:
Yeah we can do that for now, but I don't think the device driver even need to now about their session id. There are some cases like if we need it for logging and responding to /session, but we can do that in AppiumDriver.